### DISCUSSION TITLE
Prisma doesn't work with hotspot connection

### DISCUSSION THREAD

**Discussion by MilanObrenovic** [prisma/prisma#20322] [Q&A]
### Bug description

There was a hellstorm last night and the internet got knocked off in the whole country. It's been 24 hours and i am still waiting for them to fix the internet.

I am unable to continue working on the project offline without internet connection via localhost because of prisma.

Problem 1 — [NOT CONNECTED TO INTERNET] — (**internet access does NOT work**):
Prisma throws errors because it cant connect to internet. Fair enough. But why not just throw a console log error and not fetch any data in return?

Error:

![image](https://github.com/prisma/prisma/assets/56999235/53b85c00-1553-43e3-a850-5a381f045242)

Problem 2 — [CONNECTED TO 4G HOTSPOT] — (**internet access WORKS**):
When I start a hotspot connection from my iphone via 4G network and connect via my computer to my iphone hotspot, the internet works fine (i can open any other website etc), but it seems like prisma is having some strange issues, it tries fetching mongodb but it cant connect.

Error:

![image](https://github.com/prisma/prisma/assets/56999235/c0b2e92d-eda1-4832-96c5-3bca34610fd9)

### How to reproduce

1. Turn off internet on your computer
2. Add a prisma `findUnique` function somewhere in the code (try fetching from mongodb)
3. Run NextJS app

### Expected behavior

App to run and function normally but not return any data, just output error in console if there is no internet. But since i am using hotspot and have internet normally, it should run the localhost app and work normally.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Product {
  id          String     @id @default(auto()) @map("_id") @db.ObjectId
  description String
  imageUrl    String
  name        String
  price       Int
  createdAt   DateTime   @default(now())
  updatedAt   DateTime   @updatedAt
  CartItem    CartItem[]

  @@map("products")
}

model Cart {
  id        String     @id @default(auto()) @map("_id") @db.ObjectId
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
  items     CartItem[]

  @@map("carts")
}

model CartItem {
  id        String  @id @default(auto()) @map("_id") @db.ObjectId
  productId String  @db.ObjectId
  product   Product @relation(fields: [productId], references: [id], onDelete: Cascade)
  quantity  Int
  cartId    String  @db.ObjectId
  cart      Cart    @relation(fields: [cartId], references: [id], onDelete: Cascade)

  @@map("cart_items")
}

```

### Environment & setup

- OS: iMac Ventura 13.4.1 
- Database: MongoDB
- Node.js version: v18.16.1 

### Prisma Version

```json
"prisma": "^5.0.0"
```

**Comment by millsp** [comment]
Hey, thanks for the report.

> Prisma throws errors because it cant connect to internet. Fair enough. But why not just throw a console log error and not fetch any data in return?

Prisma will try to connect on the first query you emit, then will bubble up that error. Also, if the server "disappears" before a query is made, you will equally get this.

> When I start a hotspot connection from my iphone via 4G network and connect via my computer to my iphone hotspot, the internet works fine (i can open any other website etc), but it seems like prisma is having some strange issues, it tries fetching mongodb but it cant connect.

Most providers will block non-standard ports like `27017`, have you tried with a VPN on?

**Reply by MilanObrenovic** [reply]
You are suggesting me to try with a VPN on? that makes no sense. Why would a localhost connection work with a VPN, how is that possible? That would be like me (from europe) being able to connect to someone's home wifi network who lives in the US.

If the hotspot network opens websites and internet works fine, why doesnt it work for prisma localhost network?

Where is port `27017` even coming from?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
